core/src/wallet/ripple/RippleLikeAccount.cpp

```diff
@@ -461,5 +461,28 @@ namespace ledger {
             });
         }
 
+        void RippleLikeAccount::isAddressActivated(const std::string &address,
+                                                   const std::shared_ptr<api::BoolCallback> &isActivated) {
+            isAddressActivated(address).callback(getContext(), isActivated);
+        }
+
+        Future<bool> RippleLikeAccount::isAddressActivated(const std::string &address) {
+            auto xrpAddress = RippleLikeAddress::parse(address, getWallet()->getCurrency());
+            if (!xrpAddress) {
+                return Future<bool>::successful(false);
+            }
+            auto self = getSelf();
+            return _explorer->getBaseReserve().flatMap<bool>(getContext(), [self, xrpAddress](const std::shared_ptr<BigInt> &reserve) -> Future<bool> {
+                return self->_explorer->getBalance(
+                        std::vector<RippleLikeKeychain::Address>{
+                                std::dynamic_pointer_cast<RippleLikeAddress>(xrpAddress)
+                        }).map<bool>(self->getContext(), [reserve](const std::shared_ptr<BigInt> &balance) -> bool {
```

Comment:
Can you create some variables with proper name ?

```cpp
auto const isBalanceSufficient = [reserve](const std::shared_ptr<BigInt> &balance) -> bool {
   if (!reserve || !balance) {
      return false;
   }
    return reserve->compare(*balance) < 0;
}
auto const xrpAddresses = std::vector<RippleLikeKeychain::Address>{std::dynamic_pointer_cast<RippleLikeAddress(xrpAddress)}

return self->_explorer->getBalance(xrpAddresses).map<bool>(self->getContext(), isBalanceSufficient);
```

Reply:
Are we talking about readability ? If so, then we should comply to that in every call with maps ... but if we do so it is going to be less readable when you chain `map` or `flatMap` so ...

Reply:
I think the code from @KhalilBellakrid is fine and what you suggested makes everything more verbose for nothing really interesting here. Chaining futures is a very common thing to do and especially, I think your version is weird because you declare the functions in reverse order of how they will be used while the inline version from @KhalilBellakrid feels more natural to me. And it’s more functional programming-like (because of the lambda), so it’s better because giving names to things that don’t require one is always something I want to stay away.

In Haskell, we have an adage that I think can be used in _any_ language: naming is hard, so if you don’t have to name something, you’re removing a problem, especially for temporaries.

Reply:
I disagree but not completely for our case. My previous comment should not be apply for all our lambdas obviously - I only think about several layers of nested lambdas. As @phaazon said, naming is hard but not only naming. Read some codes with nested lambdas containing some advanced logics are not something I enjoy a lot. Good naming and const lambda variables increase a lot the readability of piece of codes and expose clearly what the developer -  who has written the code - want to do. 
@KhalilBellakrid This is ok the way to do and the comment you left so I'll approve this PR